2_process/src/munge_natural_baseflow.R

```diff
@@ -0,0 +1,81 @@
+munge_natural_baseflow <- function(baseflow_pred_files,segs_w_comids,vars=c("mean","med","p10","p90"),start_year,end_year){
+  #' 
+  #' @description This function reads in the DRB monthly baseflow tables from Miller et al. 2021 (https://doi.org/10.5066/P9FZG7GZ)
+  #' and outputs a data frame containing monthly baseflow estimates for each PRMS segment. The Miller et al. dataset spans
+  #' years 1950-2015. If variable {end_year} is later than 2015, monthly values will be filled with long-term monthly averages from
+  #' {start_year} to 2015. 
+  #' 
+  #' @param baseflow_pred_files file paths of unzipped monthly baseflow tables, including file extension
+  #' @param segs_w_comids data frame containing the DRB PRMS segment ids and the comids of interest
+  #' segs_w_comids must contain variables PRMS_segid and comid
+  #' @param vars columns from monthly baseflow tables to retain; can include one or all of "mean","med","p10",or "p90", 
+  #' default is to retain all four columns
+  #' @param start_year character string indicating the earliest year to return and to include in long-term average monthly baseflows
+  #' @param end_year character string indicating the latest year to return and to include in long-term average monthly baseflows
+  #'                                           
+  
+  # Read in monthly baseflow files
+  monthly_baseflow_ls <- lapply(baseflow_pred_files,function(x){
+    
+    # Extract the month for each monthly baseflow file
+    # this line finds numbers with 2 digits within the file name string
+    month <- str_extract(x, "[0-9]{2}")
+    
+    # Read in monthly tables
+    data <- read_csv(x,show_col_types = FALSE) %>%
+      # reformat column names to include month
+      rename_with(.cols = p10.Pred:mean.Pred, function(x){paste0(x,"_", month)}) %>%
```

Comment:
I've never tried making a function of x inside of a function of x. I guess the scopes are independent and it works, but I'd prefer changing to a different variable here (y?)

Reply:
Ooh, good catch! It does seem to work but I agree with you that replacing the second function call to depend on y would be easier to read and probably also safer.